api/app/Traits/Generator/GeneratesNominationDoc.php

```diff
@@ -0,0 +1,128 @@
+     *
+     * @param  Section  $section  The section to add nomination details to
+     * @param  TalentNominationGroup  $talentNominationGroup  The talent nomination group
+     */
+    protected function details(Section $section, TalentNominationGroup $talentNominationGroup, $headingRank)
+    {
+
+        $section->addTitle($this->localizeHeading('nomination_details'), $headingRank);
+
+        $nominations = $talentNominationGroup->nominations()->get();
+
+        if ($talentNominationGroup->nominations()->count() > 0) {
+            foreach ($nominations as $nomination) {
+                $section->addTitle("{$this->localizeHeading('nominated_by')} {$nomination->nominator->first_name} {$nomination->nominator->last_name}", $headingRank + 1);
+                $this->addLabelText($section, $this->localizeHeading('date_received'), $nomination->submitted_at->format('F d, Y'));
+                $this->addLabelText($section, $this->localizeHeading('nominators_work_email'), $nomination->nominator->work_email);
+                $this->addLabelText($section, $this->localizeHeading('nominators_classification'), $nomination->nominator->getClassification());
+                $this->addLabelText($section, $this->localizeHeading('nominators_department'), $nomination->nominator->department->name[$this->lang]);
+
+                $nominatorRelationship = $nomination->submitter_relationship_to_nominator ? $this->localizeEnum($nomination->submitter_relationship_to_nominator->name, TalentNominationSubmitterRelationshipToNominator::class) : $nomination->submitter_relationship_to_nominator_other;
+                $this->addLabelText($section, $this->localizeHeading('nominators_relationship'), $nominatorRelationship);
+
+                $this->addLabelText($section, $this->localizeHeading('nominated_for'), '');
+                if ($nomination->nominate_for_advancement) {
+                    $section->addListItem($this->localizeHeading('advancement'));
+                }
+                if ($nomination->nominate_for_development_programs) {
+                    $section->addListItem($this->localizeHeading('development_programs'));
+                }
+                if ($nomination->nominate_for_lateral_movement) {
+                    $section->addListItem($this->localizeHeading('lateral_movement'));
+                }
+
+                $this->addLabelText($section, $this->localizeHeading('advanced_secondary_reference'), $nomination->advancement_reference_fallback_name);
+                $this->addLabelText($section, $this->localizeHeading('references_work_email'), $nomination->advancement_reference_fallback_work_email);
+                $this->addLabelText($section, $this->localizeHeading('references_classification'), $nomination->advancementReferenceFallbackClassification->displayName);
+                $this->addLabelText($section, $this->localizeHeading('references_department'), $nomination->advancementReferenceFallbackDepartment->name[$this->lang]);
```

Comment:
Handling of different nomination types failing in some cases. 
For example, when downloading where advancement is not applicable I get an error

<img width="1861" height="117" alt="image" src="https://github.com/user-attachments/assets/8b3b505a-4849-4086-b938-31be35725b5c" />

Reply:
Thanks for catching that it should be resolved now